src/services/guidanceresponse.service.ts

```diff
@@ -1,86 +1,14 @@
-import { FhirUtilities } from '../fhir/utilities';
-import GuidanceResponseModel from '../lib/schemas/resources/GuidanceResponse';
-import { Parameters, Medication, Patient, MedicationRequest } from 'fhir/r4';
-import config from '../config';
-import axios from 'axios';
+// TODO: This operation should forward the request to the actual server using the phonebook. 
 
 module.exports.searchById = async (args: any) => {
-  const { id } = args;
-  console.log('GuidanceResponse >>> searchById: -- ' + id);
-  return await GuidanceResponseModel.findOne({ id: id.toString() }, { _id: 0 }).exec();
 };
 
 module.exports.create = async (args: any, req: any) => {
-  console.log('GuidanceResponse >>> create');
-  const resource = req.req.body;
-  const { base_version } = args;
-  return await FhirUtilities.store(resource, GuidanceResponseModel, base_version);
 };
 
-const getMedicationCode = (
-  medication: Medication | MedicationRequest | undefined
-): string | undefined => {
-  // grab the medication drug code from the Medication resource
-  let drugCode;
-  if (medication?.resourceType == 'Medication') {
-    medication?.code?.coding?.forEach(medCode => {
-      if (medCode?.system?.endsWith('rxnorm')) {
-        drugCode = medCode?.code;
-      }
-    });
-  } else {
-    if (medication?.medicationCodeableConcept) {
-      medication?.medicationCodeableConcept?.coding?.forEach(medCode => {
-        if (medCode.system?.endsWith('rxnorm')) {
-          drugCode = medCode.code;
-        }
-      });
-    } else if (medication?.medicationReference) {
-      const ref = medication.medicationReference.reference;
-      if (ref?.startsWith('#')) {
-        const containedRef = ref.slice(1);
-        const match = medication.contained?.find(res => {
-          return res.id === containedRef;
-        });
-        if (match?.resourceType === 'Medication') {
-          return getMedicationCode(match);
-        }
-      }
-    }
-  }
-  return drugCode;
-};
 
 module.exports.remsEtasu = async (args: any, context: any, logger: any) => {
```

Comment:
Should we remove all of this code? The $rems-etasu check was working before but just forwarding to a single rems-admin. After this change it completely removes the functionality until it can be added back in with another task. I'd like to see the simple functionality remain so the implementer of the other feature can base it upon what was in here to begin with.

Reply:
I don't think any of this code is relevant anymore. It seems like the reason it was working was due to the url for the rems admin being included in the config directly and used by this code. Also, most of the functions used here no longer exist in this repo, so we'd have to comment most of this code out, at which point I'd just encourage future implementers to look at the git history. But regardless, everything that happens here should be happening at the rems admin, right?

Reply:
You are right that most of it might not have been needed anymore. The part below (lines 74 - 85) were necessary to forward the ETASU call. It was only working with a single configured rems-admin, but it allowed for some testing. Without this code the implementer of the ETASU changes will need to look back at the history to find this.